notebooks/metatranscriptomics_pipeline_iSeqnorm.ipynb

```diff
@@ -0,0 +1,1477 @@
+   "source": [
+    "picklist = format_pooling_echo_pick_list(vols, max_vol_per_well=30000)\n",
+    "with open(picklist_fp,'w') as f:\n",
+    "    f.write(picklist)\n",
+    "\n",
+    "!head {picklist_fp}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Write DataFrame to file\n",
+    "\n",
+    "We want to keep all that useful information together in one place so it can be easily parsed later. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Write the sample info DataFrame as .txt\n",
+    "plate_df_fp = './test_output/QC/YYYY_MM_DD_NPH_3-6_df.txt'\n",
+    "\n",
+    "if os.path.isfile(plate_df_fp):\n",
+    "    print(\"Warning! This file exists already.\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plate_df.to_csv(plate_df_fp, sep='\\t')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true,
+    "jupyter": {
+     "outputs_hidden": true
+    }
+   },
+   "source": [
+    "# Make sample sheet\n",
+    "\n",
+    "This workflow takes the pooled sample information and writes an Illumina sample sheet that can be given directly to the sequencing center. \n",
+    "\n",
+    "As inputs, this notebook requires:\n",
+    "1. A plate map DataFrame (from previous step)\n",
+    "\n",
+    "The workflow:\n",
+    "1. formats sample names as bcl2fastq-compatible\n",
+    "2. formats sample data\n",
+    "3. sets values for sample sheet fields and formats sample sheet.\n",
+    "4. writes the sample sheet to a file"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 1: Format sample names to be bcl2fastq-compatible\n",
+    "\n",
+    "bcl2fastq requires *only* alphanumeric, hyphens, and underscore characters. We'll replace all non-those characters\n",
+    "with underscores and add the bcl2fastq-compatible names to the DataFrame."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plate_df['sample sheet Sample_ID'] = plate_df['Sample'].map(bcl_scrub_name)\n",
+    "\n",
+    "plate_df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 2: format sample sheet data\n",
+    "\n",
+    "This step formats the data columns appropriately for the sample sheet, using the values we've calculated previously.\n",
+    "\n",
+    "The newly-created bcl2fastq-compatible names will be in the **`Sample ID`** and **`Sample Name`** columns. The\n",
+    "original sample names will be in the **`Description`** column.\n",
+    "\n",
+    "Modify **`lanes`** to indicate which lanes this pool will be sequenced on.\n",
+    "\n",
+    "**Project Name and Project Plate values will be placed in the **`Sample_Project`** and **`Sample_Name`**\n",
```

Comment:
Fix formatting: the `**` for bolding is opened right before `Project Name` but not closed, thus causing the `**` later for `Sample_Project` to read wrong.